pkg/profile/profile.go

```diff
@@ -45,20 +57,38 @@ func New(opt *option.Options) (Profile, error) {
 		opt: opt,
 	}
 
-	err := p.startCPUProfile()
+	err := p.StartCPUProfile("")
 	if err != nil {
 		return nil, err
 	}
 
 	return p, nil
 }
 
-func (p *profile) startCPUProfile() error {
-	if p.opt.CPUProfileFile == "" {
+func (p *profile) CPUFileName() string {
+	if p.cpuFile == nil {
+		return ""
+	}
+	return p.cpuFile.Name()
+}
+
+func (p *profile) MemoryFileName() string {
+	return p.memFileName
+}
+
+func (p *profile) UpdateMemoryProfile(fp string) {
```

Comment:
There could be concurrent calls from the network, we need to handle it, don't we? So do others.

Reply:
I think that we can suppose that there is only one administrator, who wants to turn on/off profiling at once.. However do you think that concurrent calls can cause runtime errors? I'll add the lock like in other admin APIs.

Reply:
We can't make assumptions from users. Runtime errors would be many kinds of race conditions.

Reply:
Yeah true.. Now it should be better, with lock for modifications.